examples/test-player.rs

```diff
@@ -0,0 +1,465 @@
+            fn metadata_loaded(&self, metadata: Metadata) {
+                self.sender
+                    .send(PlayerEvent::MetadataLoaded(metadata))
+                    .unwrap();
+            }
+            fn duration_changed(&self, _duration: f64) {}
+            fn loaded_data(&self) {}
+            fn time_update(&self, _time: f64) {}
+            fn seek_started(&self) {}
+            fn seek_completed(&self) {}
+            fn update_current_images(&self, images: Vec<PlanarYCbCrImage>) {
+                self.sender
+                    .send(PlayerEvent::UpdateCurrentImages(images))
+                    .unwrap();
+            }
+            fn buffered(&self, ranges: Vec<Range<f64>>) {
+                self.sender
+                    .send(PlayerEvent::BufferedRanges(ranges))
+                    .unwrap();
+            }
+            fn seekable(&self, ranges: Vec<Range<f64>>) {
+                self.sender
+                    .send(PlayerEvent::SeekableRanges(ranges))
+                    .unwrap();
+            }
+        }
+
+        let (shutdown_sender, shutdown_receiver) = mpsc::channel();
+        let wrapper_sender = sender.clone();
+        thread::spawn(move || {
+            let sink = Box::new(Sink { sender: sender });
+            let player = GeckoMedia::get()
+                .unwrap()
+                .create_blob_player(bytes, mime, sink)
+                .unwrap();
+            player.set_volume(1.0);
+            loop {
+                match receiver.recv().unwrap() {
+                    PlayerEvent::Ended => {
+                        println!("Ended");
+                        unsafe {
+                            CLOSE_WINDOW = true;
+                        };
+                        break;
+                    }
+                    PlayerEvent::Error => {
+                        println!("Error");
+                        break;
+                    }
+                    PlayerEvent::AsyncEvent(name) => {
+                        println!("Event received: {:?}", name);
+                    }
+                    PlayerEvent::MetadataLoaded(metadata) => {
+                        println!("MetadataLoaded; duration: {:?}", metadata.duration);
+                    }
+                    PlayerEvent::StartPlayback => {
+                        player.play();
+                    }
+                    PlayerEvent::UpdateCurrentImages(images) => {
+                        frame_sender.send(images).unwrap();
+                    }
+                    PlayerEvent::BreakOutOfEventLoop => {
+                        break;
+                    }
+                    PlayerEvent::BufferedRanges(ranges) => {
+                        println!("Buffered ranges: {:?}", ranges);
+                    }
+                    PlayerEvent::SeekableRanges(ranges) => {
+                        println!("Seekable ranges: {:?}", ranges);
+                    }
+                };
+            }
+            drop(player);
```

Comment:
Why is this drop needed?

Reply:
This drop ensures that the Player object is destroyed before we send the shutdown notification. The other thread blocks waiting for the confirmation that the Player is shutdown before shutting down GeckoMedia. If we don't drop here, the Player may not have shutdown when the other thread tries to shutdown GeckoMedia, which will assert if there are Player objects alive.

Reply:
I'll put the above explanation in a comment.